charts/redpanda/statefulset.go

```diff
@@ -247,3 +247,290 @@ func StatefulSetVolumeMounts(dot *helmette.Dot) []corev1.VolumeMount {
+		Image:   fmt.Sprintf("%s:%s", values.Image.Repository, Tag(dot)),
+		Command: []string{`/bin/sh`},
+		Args: []string{
+			`-c`,
+			fmt.Sprintf(`trap "exit 0" TERM; exec /etc/secrets/fs-validator/scripts/fsValidator.sh %s & wait $!`,
+				values.Statefulset.InitContainers.FSValidator.ExpectedFS,
+			),
+		},
+		SecurityContext: ptr.To(ContainerSecurityContext(dot)),
+		VolumeMounts: append(append(CommonMounts(dot),
+			values.Statefulset.InitContainers.FSValidator.ExtraVolumeMounts...),
+			corev1.VolumeMount{
+				Name:      fmt.Sprintf(`%.49s-fs-validator`, Fullname(dot)),
+				MountPath: `/etc/secrets/fs-validator/scripts/`,
+			},
+			corev1.VolumeMount{
+				Name:      `datadir`,
+				MountPath: `/var/lib/redpanda/data`,
+			},
+		),
+		Resources: helmette.UnmarshalInto[corev1.ResourceRequirements](values.Statefulset.InitContainers.FSValidator.Resources),
+	}
+}
+
+func statefulSetInitContainerSetTieredStorageCacheDirOwnership(dot *helmette.Dot) *corev1.Container {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if !storageTieredEnabled(dot) {
+		return nil
+	}
+
+	uid, gid := securityContextUidGid(dot, "set-tiered-storage-cache-dir-ownership")
+	cacheDir := storageTieredCacheDirectory(dot)
+	mounts := CommonMounts(dot)
+	mounts = append(mounts, corev1.VolumeMount{
+		Name:      "datadir",
+		MountPath: "/var/lib/redpanda/data",
+	})
+	if storageTieredMountType(dot) != "none" {
+		name := "tiered-storage-dir"
+		if values.Storage.PersistentVolume != nil && values.Storage.PersistentVolume.NameOverwrite != "" {
+			name = values.Storage.PersistentVolume.NameOverwrite
+		}
+		mounts = append(mounts, corev1.VolumeMount{
+			Name:      name,
+			MountPath: cacheDir,
+		})
+	}
+	mounts = append(mounts, values.Statefulset.InitContainers.SetTieredStorageCacheDirOwnership.ExtraVolumeMounts...)
+
+	return &corev1.Container{
+		Name:  `set-tiered-storage-cache-dir-ownership`,
+		Image: fmt.Sprintf(`%s:%s`, values.Statefulset.InitContainerImage.Repository, values.Statefulset.InitContainerImage.Tag),
+		Command: []string{
+			`/bin/sh`,
+			`-c`,
+			fmt.Sprintf(`mkdir -p %s; chown %d:%d -R %s`,
+				cacheDir,
+				uid, gid,
+				cacheDir,
+			),
+		},
+		VolumeMounts: mounts,
+		Resources:    helmette.UnmarshalInto[corev1.ResourceRequirements](values.Statefulset.InitContainers.SetTieredStorageCacheDirOwnership.Resources),
+	}
+}
+
+// storageTieredConfig was: storage-tiered-config:
+// Support legacy storage.tieredConfig
+func storageTieredConfig(dot *helmette.Dot) TieredStorageConfig {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if values.Storage.TieredConfig != nil {
+		return values.Storage.TieredConfig
+	}
+	return values.Storage.Tiered.Config
+}
+
+// storageTieredEnabled - check whether tiered storage is enabled
+func storageTieredEnabled(dot *helmette.Dot) bool {
+	config := storageTieredConfig(dot)
+
+	enabled := helmette.Dig(config, "/var/lib/redpanda/data/cloud_storage_cache", `cloud_storage_enabled`).(bool)
+	return enabled
+}
```

Comment:
NIT: The following method could be used https://github.com/redpanda-data/helm-charts/blob/d2444af416d0e182f476bb231f90d7b2b7c636d7/charts/redpanda/values.go#L368-L373

Reply:
Why the `"/var/lib/redpanda/data/cloud_storage_cache"` is passed as fallback, but then it is type assertion of `(bool)`?